src/dbally/similarity/chroma_store.py

```diff
@@ -0,0 +1,96 @@
+from typing import List, Literal, Optional, Union
+
+import chromadb
+
+from dbally.embedding_client.base import EmbeddingClient
+from dbally.similarity.store import SimilarityStore
+
+
+class ChromadbStore(SimilarityStore):
+    """Class that stores text embeddings using [Chroma](https://docs.trychroma.com/)"""
+
+    def __init__(
+        self,
+        index_name: str,
+        client: chromadb.Client,
+        embedding_calculator: Union[EmbeddingClient, chromadb.EmbeddingFunction],
+        max_distance: Optional[float] = None,
+        distance_method: Literal["l2", "ip", "cosine"] = "l2",
+    ):
+        super().__init__()
+        self.index_name = index_name
+        self.client = client
+        self.embedding_calculator = embedding_calculator
+        self.max_distance = max_distance
+
+        self._metadata = {"hnsw:space": distance_method}
+
+    def _get_collection(self) -> chromadb.Collection:
+        """Based on the selected embedding_calculator chooses how to retrieve the collection.
+            If collection doesn't exist it creates one.
+
+        Returns:
+            chromadb.Collection: Retrieved collection
+        """
+        if isinstance(self.embedding_calculator, EmbeddingClient):
+            return self.client.get_or_create_collection(name=self.index_name, metadata=self._metadata)
+
+        return self.client.get_or_create_collection(
+            name=self.index_name, metadata=self._metadata, embedding_function=self.embedding_calculator
+        )
+
+    def _return_best_match(self, retrieved: dict) -> Optional[str]:
+        """Based on the retrieved data returns the best match or None if no match is found.
+
+        Args:
+            retrieved: Retrieved data, with a column first format
+
+        Returns:
+            The best match or None if no match is found
+        """
+        if self.max_distance is None or retrieved["distances"][0][0] <= self.max_distance:
+            return retrieved["documents"][0][0]
+
+        return None
+
+    async def store(self, data: List[str]) -> None:
+        """
+        Fills chroma collection with embeddings of provided string. As the id uses hash value of the string.
+
+        Args:
+            data: The data to store.
+        """
+
+        # chroma requires an unique id for every document
```

Comment:
If chroma requires an **unique** id for every document than the `hash` function is not the right solution:

1. It will produce the same hash each time the items happen to have the same contents
2. It does not guarantee no collisions and its space is relatively small 

You should probably generate uuids, or if you want hash (and the first point isn't a concern) - a proper hash function like sha256.

Reply:
> 1. It will produce the same hash each time the items happen to have the same contents

It is actually desired behaviour for us - if contents are the same we're fine with using the same document

agree with sha256

Reply:
> It is actually desired behaviour for us - if contents are the same we're fine with using the same document

I'm just going by the "requires an unique id for every document" comment. If it doesn't, the comment should probably by changed instead.